src/cc65/expr.c

```diff
@@ -2501,6 +2501,37 @@ static void hie_internal (const GenDesc* Ops,   /* List of generators */
                 }
             }
 
+            /* Limit the calculated value to the range of its type */
+            switch (GetUnderlyingTypeCode (Expr->Type)) {
+                case T_INT:
+                case T_SHORT:
+                    Expr->IVal = (signed short)Expr->IVal;
```

Comment:
Nit: Maybe `CHECK (sizeof (short) == 2)`, unless this is a global assumption.

Reply:
This should be a global static assertion, together with `sizeof (long) == 4`.

Reply:
Apparently, this is supposed to compile with C89, so C11 is a bit out of the question.

I'm going to try to add `stdint.h`-like typedefs to `common/inttypes.h` and see if that gets merged. Then this can just be `int16_t`.

Reply:
> This should be a global static assertion, together with `sizeof (long) == 4`.

I've been assuming the assumption was `sizeof(long) >= 4`. `sizeof(long) == 8` on most 64-bit platforms, AFAIK. Using `int32_t` would make this clear. #1326 adds these for C89 compilers.

Reply:
Resolved.